src/specfem3D/setup_sources_receivers.f90

```diff
@@ -123,6 +123,9 @@ subroutine setup_sources()
     allocate(pm1_source_encoding(1),stat=ier)
   endif
   if (ier /= 0) stop 'error allocating arrays for sources'
+  
+  allocate(user_source_time_function(NSTEP, NSOURCES),stat=ier)
```

Comment:
hi Vadim, 
better if this would only get allocated in case EXTERNAL_STF is actually used. the array can get very big for cases with many sources (kinematic rupture simulations) and/or longer simulation times. 

if EXTERNAL_STF is not used, either allocate a dummy array (user_source_time_function(1,1)) or don't allocate at all and remove if from subroutine argument calls (put it in the "use specfem_par,only: .." part).